test/e2e/redundancy/gateway_failover.go

```diff
@@ -87,7 +87,8 @@ func testGatewayPodRestartScenario(f *subFramework.Framework) {
 
 	framework.By(fmt.Sprintf("Ensuring that the gateway reports as active on %q", primaryClusterName))
 
-	activeGateway := f.AwaitGatewayFullyConnected(primaryCluster, resource.EnsureValidName(gatewayNodes[0].Name))
+	submEndpoint := f.AwaitSubmarinerEndpoint(primaryCluster, subFramework.NoopCheckEndpoint)
```

Comment:
It is corner case, but it better check that  submEndpoint  is not nil

Expect(submEndpoint).ToNot(BeNil(), "Did not find local endpoint")

Reply:
It's actually not necessary b/c AwaitSubmarinerEndpoint will itself fail the test if it's not found. Also note that other call sites don't check for nil.